menu.py

```diff
@@ -100,39 +126,259 @@ def key_press(self, key: Key) -> Menu | None:
                     selected_menu.call()
                     return_menu = self
                     if old_name != selected_menu.display_str:
-                        self.display.update_row(self.selected_on_display, selected_menu.display_str)
+                        self.display.update_row(
+                            self._display_row(self.selected),
+                            selected_menu.display_str,
+                            highlight=True
+                        )
             else:
                 return_menu = self
 
         return return_menu
 
     def redraw(self) -> None:
-        self.display.clear()
-        self.display.highlight_text(self.selected_on_display)
-        start_from = max(0, self.selected - self.selected_on_display)
-        for index in range(start_from, min(len(self.menu_elements), self.display.rows + start_from)):
-            self.display.push_back(self.menu_elements[index].display_str)
-
-class MenuRoot:
-    def __init__(self, root: Menu, display: BaseDisplayHandler = BaseDisplayHandler()) -> None:
-        self.root = root
-        self.current_menu: Menu | None = None
-        self.root.set_display(display)
-
-    def key_press(self, key: Key) -> bool:
-        """ @return true if menu closed """
-        if self.current_menu is None:
-            self.current_menu = self.root
-            self.current_menu.redraw()
-        else:
-            self.current_menu = self.current_menu.key_press(key)
-        return self.current_menu is None
+        display_str = [
+            menu.display_str
+            for menu in self.menu_elements[self.start_row : self.start_row + self.display.rows]
```

Comment:
`self.start_row:` (remove whitespace)
Please consider.

Reply:
I think it's more readable when using complex expressions in slice.
https://peps.python.org/pep-0008/#whitespace-in-expressions-and-statements
```
# Correct:
ham[1:9], ham[1:9:3], ham[:9:3], ham[1::3], ham[1:9:]
ham[lower:upper], ham[lower:upper:], ham[lower::step]
ham[lower+offset : upper+offset]
ham[: upper_fn(x) : step_fn(x)], ham[:: step_fn(x)]
ham[lower + offset : upper + offset]
```

Reply:
Agree, but we don't have a complex expression on left.

Reply:
But we have on the right and i think it's more readable like that. And it's correct.
I'll quote PEP8 again:
```python
# Correct:
ham[lower+offset : upper+offset]
ham[lower + offset : upper + offset]
# Wrong:
ham[lower + offset:upper + offset]
```

Reply:
I will leave that as open, E203. Removing left whitespace resolves the issue.

Reply:
https://github.com/PyCQA/pycodestyle/issues/373